### PR TITLE
update to guava 32 and orbit 2023-09

### PR DESCRIPTION
**Pull request by cdietrich**
update to guava 32 and orbit 2023-09

### CODE REVIEW

File: gradle/versions.gradle
Comment (cdietrich) [inline gradle/versions.gradle:16]: am not sure how to properly chickend and egg bootstrap this? use range?

**Comment by cdietrich** [comment]
is there no jenkins anymore?

**Comment by jonahgraham** [comment]
> is there no jenkins anymore?

Not for PRs, but for branches. It looks like it is hung thought - https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/cd_guavaBmp/1/console

File: gradle/versions.gradle
Comment (jonahgraham) [reply gradle/versions.gradle:16]: Is this a typo? Why is this version older?

Comment (jonahgraham) [reply gradle/versions.gradle:16]: I'm guess a typo as 2.**32**.0.M2 was [published a few days ago](https://central.sonatype.com/artifact/org.eclipse.xtend/org.eclipse.xtend.core/2.32.0.M2)

A bit concerning that the build can succeed without this being checked.

Comment (jonahgraham) [reply gradle/versions.gradle:16]: > A bit concerning that the build can succeed without this being checked.

(Sorry for stream of consciousness as I learn what is going on)

It probably succeeded because there is a 2.23.0.M2 in maven central too.

```suggestion
	'xtend_lib': '2.32.0.M2',

```

File: gradle/versions.gradle
Comment (jonahgraham) [reply gradle/versions.gradle:18]: I don't know.

File: releng/releng-target/lsp4j.target.target
Comment (jonahgraham) [inline releng/releng-target/lsp4j.target.target:7]: To use the Maven Central version of Gson in OSGi context we needed #738 fixed, so I have gone ahead and completed that too.

**Comment by jonahgraham** [comment]
> > is there no jenkins anymore?
> 
> Not for PRs, but for branches. It looks like it is hung though - https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/cd_guavaBmp/1/console

It looks like a transient Jenkins issue - [job 3](https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/cd_guavaBmp/3/console) worked fine and is reporting back to the here successfully.

**Review by jonahgraham — APPROVED** [review]
Approved - before I submit I guess we need to decide about bumping min version. I applied the xtend_lib change to your branch.

**Comment by cdietrich** [comment]
thx for fixing typo

File: gradle/versions.gradle
Comment (cdietrich) [reply gradle/versions.gradle:18]: as we use it in generator only i would propose to
the question is more about lsp4e (? am not sure if i remember correct?) we have the gson range for

Comment (jonahgraham) [reply gradle/versions.gradle:18]: > as we use it in generator only i would propose to

+1

> the question is more about lsp4e (? am not sure if i remember correct?) we have the gson range for

Sorry - not sure I follow what you are asking here.

Comment (jonahgraham) [reply gradle/versions.gradle:18]: Ahh - I see. Anyway, in light of CVE (as mentioned in https://github.com/eclipse-lsp4j/lsp4j/pull/750#issue-1836766128) I guess we should just make that the minimum.

Comment (jonahgraham) [reply gradle/versions.gradle:18]: I'll commit this:

```suggestion
	'guava': '[32.1.1,33)',
```

File: gradle/versions.gradle
Comment (jonahgraham) [inline gradle/versions.gradle:18]: For some reason suggestion is defaulting to adding a newline at the end - this suggestion removes the one I accidentally added earlier.

```suggestion
	'xtend_lib': '2.32.0.M2',
	'guava': '[32.1.1,33)',
```

**Comment by jonahgraham** [comment]
Once this builds successfully I will merge it assuming no objections.

**Comment by jonahgraham** [comment]
The macOS Jakarta websocket tests seem unstable. Not sure if this is a new instability, but they have failed ~25% of the time.

**Pull request merged**
